### PR TITLE
issue 560: updates DateInput component's readOnly prop to be configurable by consumer implementation

### PR DESCRIPTION
**Pull request by ViktoriyaSavkina**
This PR solves issue 560 as per discussion there
https://github.com/airbnb/react-dates/issues/560

### CODE REVIEW

**Comment by majapw** [comment]
Please rebase and fix the corresponding test. O/w looks good to me! :D

**Comment by ViktoriyaSavkina** [comment]
@majapw done :)

File: src/components/DateInput.jsx [hunk 1]
```diff
@@ -131,7 +131,6 @@ export default class DateInput extends React.Component {
   render() {
     const {
       dateString,
-      isTouchDevice: isTouch,
```
Comment (ljharb) [inline src/components/DateInput.jsx]: do we still need the `isTouchDevice` state?

File: src/components/DateInput.jsx [hunk 2]
```diff
@@ -172,7 +171,7 @@ export default class DateInput extends React.Component {
           placeholder={placeholder}
           autoComplete="off"
           disabled={disabled}
-          readOnly={readOnly || isTouch}
+          readOnly={readOnly}
```
Comment (ljharb) [inline src/components/DateInput.jsx]: i wonder if we could make this semver-patch by changing this to `typeof readOnly === 'boolean' ? ‼readOnly : isTouch`, and making the `defaultProp` be `null`?

**Comment by ViktoriyaSavkina** [comment]
@ljharb updated per your comments

**Comment by ljharb** [comment]
we shouldn't fear breaking changes, but we should strive to avoid them :-)

as for coveralls, it might be having issues on all repos; i've seen coverage badges broken for awhile on a bunch of them. i'm not sure what the "unknown coverage" is for tho.

**Pull request merged**